### PR TITLE
Initial support for Android in main CMake build

### PR DESCRIPTION
**Pull request by juan-lunarg**
- Minor C++ code fixes
- Minor fix to CMake to address finding things when cross-compiling
- Add minimal CI

### CODE REVIEW

**Comment by juan-lunarg** [comment]
Currently testing doesn't compile. That will be the next thing to address.

**Review by spencer-lunarg — APPROVED** [review]

File: .github/workflows/vvl.yml [hunk 1]
```diff
@@ -215,6 +215,31 @@ jobs:
...
+        - uses: lukka/get-cmake@latest
+        - name: Setup ccache
+          uses: hendrikmuhs/ccache-action@v1.2
+          with:
+            key: ${{ runner.os }}-android-cmake
+        - name: Add ccache to PATH
+          run: echo "/usr/lib/ccache" >> $GITHUB_PATH
+        - name: Configure
+          run: |
+            cmake -S . -B build/ --toolchain $ANDROID_NDK_ROOT/build/cmake/android.toolchain.cmake \
```
Comment (juan-lunarg) [inline .github/workflows/vvl.yml:234]: https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#android

1.) The Android NDK comes pre-installed
2.) `ANDROID_NDK_ROOT` is set by the docker image for us

**Review by ncesario-lunarg — APPROVED** [review]
:scream: are these really the only changes that are necessary? If so, that would be truly awesome!

**Comment by juan-lunarg** [comment]
> scream are these really the only changes that are necessary? If so, that would be truly awesome!

The hard part is going to be untangling the old systems. Actually adding the android support to the main CMake build isn't that hard.

**Pull request merged**